functions/fish_prompt.fish

```diff
@@ -846,6 +847,85 @@ function __bobthefish_prompt_rubies -S -d 'Display current Ruby information'
     echo -ns $ruby_glyph $ruby_version ' '
 end
 
+function __bobthefish_prompt_golang -S -d 'Display current Go information'
+    [ "$theme_display_go" = 'no' ]
+    and return
+    
+    set -l go_version
+    if type -fq asdf
+        set -l asdf_golang_version (asdf current golang 2>/dev/null)
+        or set -l asdf_golang_version "na"
+        
+        set -l asdf_go_sdk_version (asdf current go-sdk 2>/dev/null)
```

Comment:
if we already got a value above, there's no need to check here, right? save ourselves the `asdf` call?

it sacrifices a tiny bit of readability, but saves a decent amount of complexity to inline all the things :) 

```fish
if type -fq asdf
  asdf current golang 2>/dev/null | read -l __ asdf_go_version __
  or asdf current go-sdk 2>/dev/null | read -l __ asdf_go_version __
  and set go_version $asdf_go_version
end
```

Reply:
oh, sorry, i didn't realize you were still changing things :)